diff --git a/client/src/main/java/com/influxdb/client/internal/InvokableScriptsApiImpl.java b/client/src/main/java/com/influxdb/client/internal/InvokableScriptsApiImpl.java
index 8f66707f70..cd96ad47ad 100644
--- a/client/src/main/java/com/influxdb/client/internal/InvokableScriptsApiImpl.java
+++ b/client/src/main/java/com/influxdb/client/internal/InvokableScriptsApiImpl.java
@@ -94,7 +94,7 @@ final class InvokableScriptsApiImpl extends AbstractQueryApi implements Invokabl
 
         Arguments.checkNotNull(query, "query");
 
-        Call<Scripts> call = service.getScripts(query.getLimit(), query.getOffset());
+        Call<Scripts> call = service.getScripts(query.getLimit(), query.getOffset(), null, null, null);
 
         return execute(call).getScripts();
     }
diff --git a/client/src/main/java/com/influxdb/client/internal/UsersApiImpl.java b/client/src/main/java/com/influxdb/client/internal/UsersApiImpl.java
index b841ac27c2..8be6a7eb82 100644
--- a/client/src/main/java/com/influxdb/client/internal/UsersApiImpl.java
+++ b/client/src/main/java/com/influxdb/client/internal/UsersApiImpl.java
@@ -96,7 +96,6 @@ final class UsersApiImpl extends AbstractRestClient implements UsersApi {
         Arguments.checkNotNull(user, "User");
 
         PostUser request = new PostUser()
-                .oauthID(user.getOauthID())
                 .name(user.getName())
                 .status(PostUser.StatusEnum.fromValue(user.getStatus().getValue()));
 
@@ -112,7 +111,6 @@ final class UsersApiImpl extends AbstractRestClient implements UsersApi {
         Arguments.checkNotNull(user, "User");
 
         PostUser request = new PostUser()
-                .oauthID(user.getOauthID())
                 .name(user.getName())
                 .status(PostUser.StatusEnum.fromValue(user.getStatus().getValue()));
 
