diff --git a/Client/UsersApi.cs b/Client/UsersApi.cs
index bb4899a..7f82854 100644
--- a/Client/UsersApi.cs
+++ b/Client/UsersApi.cs
@@ -366,7 +366,7 @@ namespace InfluxDB.Client
         private PostUser ToPostUser(User user)
         {
             Enum.TryParse(user.Status.ToString(), true, out PostUser.StatusEnum status);
-            var postUser = new PostUser(user.OauthID, user.Name, status);
+            var postUser = new PostUser(user.Name, status);
             return postUser;
         }
     }
